provisioning_templates/snippet/kickstart_kernel_options.erb

```diff
@@ -0,0 +1,108 @@
+
+  # networking credentials
+  raise("Dual-stack provisioning not supported") if subnet4 && subnet6
+  if subnet4 && subnet4.dhcp_boot_mode?
+    options.push("ip=dhcp") if rhel_compatible && major >= 7
+  elsif subnet4 && !subnet4.dhcp_boot_mode?
+    if rhel_compatible && major < 7
+      dns_servers = subnet4.dns_servers.join(',')
+      if @ipxe_net
+        options.push('ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}')
+      else
+        options.push("ip=#{iface.ip}", "netmask=#{subnet4.mask}", "gateway=#{subnet4.gateway}", "dns=#{dns_servers}")
+      end
+    else
+      if @ipxe_net
+        options.push("ip=${netX/ip}::${netX/gateway}:${netX/netmask}:#{hostname}:#{iface.identifier}:none nameserver=${dns}")
+      else
+        options.push("ip=#{iface.ip}::#{subnet4.gateway}:#{subnet4.mask}:#{hostname}:#{iface.identifier}:none")
+        subnet4.dns_servers.each { |server|
+          options.push("nameserver=#{server}")
+        }
+        end
+    end
+  elsif subnet6.dhcp_boot_mode? && rhel_compatible && major >= 7
+    if host_param_true?("disable-slaac")
+      options.push("ip=dhcp6")
+    else
+      options.push("ip=auto6")
+    end
+  elsif subnet6 && !subnet6.dhcp_boot_mode?
+    raise("Static IPv6 provisioning works on RHEL7 or newer") if rhel_compatible && major < 7
+    gw = subnet6.gateway ? "[#{subnet6.gateway}]" : ""
+    options.push("ip=[#{iface.ip6}]::#{gw}:#{subnet6.cidr}:#{hostname}:#{iface.identifier}:none")
+    subnet6.dns_servers.each { |server|
+      options.push("nameserver=[#{server}]")
+    }
+  end
+
+  # bond
+  if iface.bond? && rhel_compatible && os_major >= 6
+    options.push("bond=#{iface.identifier}:#{iface.attached_devices_identifiers}:mode=#{iface.mode},#{iface.bond_options.tr(' ', ',')}")
+  end
+
+  # VLAN (only on physical is recognized)
+  if iface.virtual? && iface.tag.present? && iface.attached_to.present?
+    if (is_fedora && os_major < 17) || (rhel_compatible && os_major < 7)
+      options.push("vlanid=#{iface.tag}")
+    else
+      options.push("vlan=vlan#{iface.tag}:#{iface.attached_to}")
+    end
+  end
+
+  # optional repository for Atomic
+  if @host.operatingsystem.name.match(/Atomic/i)
+    options.push("inst.repo=#{medium_uri}")
+  end
+
+  if @host.operatingsystem.name.match(/Atomic/i) || host_param('kickstart_liveimg')
+    options.push("inst.stage2=#{medium_uri}")
+  end
+
+  if !is_fedora && os_major >= 7 && host_param_true?('fips_enabled')
+    options.push('fips=1')
+  end
+
+  nic_delay = subnet4 ? subnet4.nic_delay : nil
+  nic_delay ||= subnet6 ? subnet6.nic_delay : nil
+  if nic_delay && rhel_compatible && major >= 7
+    ["dhcp", "iflink", "ifup", "route", "ipv6dad", "ipv6auto", "carrier"].each do |type|
+      options.push("rd.net.timeout.#{type}=#{nic_delay}")
+    end
+  elsif nic_delay
+    options.push("nicdelay=#{nic_delay} linksleep=#{nic_delay}")
+  end
+-%>
+<%= options.join(' ') -%>
```

Comment:
Is the lack of a newline intentional?

Reply:
Yes, this is added to another line actually, it's "line snippet". I've tested this and it renders fine.

Reply:
If you rely on it, I'd add a comment above it. It's too easy to miss in doing a review. Alternatively you can perhaps call `snippet().strip()` when using this if it's important?

Reply:
Well as long as this is added as the very last element on that line it actually does not matter, I don't think it is worth the effort, it's pretty obvious when this is broken.

Reply:
Always adding it as the last element so the newline doesn't matter is another valid technical solution.

Reply:
Added comment.